tests/post/steps/test_salt_api.py

```diff
@@ -54,30 +65,49 @@ def context():
     "we login to SaltAPI as '{username}' using password '{password}'"))
 def login_salt_api_basic(host, username, password, version, context):
     address = _get_salt_api_address(host, version)
-    token = base64.encodebytes(
-        '{}:{}'.format(username, password).encode('utf-8')
-    ).rstrip()
     context['salt-api'] = _salt_api_login(
-        address, username=username, password=token
+        address, username=username, password=password
     )
 
 
 @when(parsers.parse(
-    "we login to SaltAPI with the ServiceAccount '{account_name}'"))
-def login_salt_api_token(host, k8s_client, account_name, version, context):
+    "we login to SaltAPI with the ServiceAccount '{namespace}/{account_name}'"))
+def login_salt_api_token(
+    host, k8s_client, namespace, account_name, version, context
+):
     address = _get_salt_api_address(host, version)
     service_account = k8s_client.read_namespaced_service_account(
-        name=account_name, namespace='kube-system'
+        name=account_name, namespace=namespace
     )
+    username = 'system:serviceaccount:{}:{}'.format(namespace, account_name)
     secret = k8s_client.read_namespaced_secret(
-        name=service_account.secrets[0].name, namespace='kube-system'
+        name=service_account.secrets[0].name, namespace=namespace
     )
     token = base64.decodebytes(secret.data['token'].encode('utf-8'))
     context['salt-api'] = _salt_api_login(
-        address, username=account_name, token=token
+        address, username=username, token=token
     )
 
 
+@when(parsers.parse(
```

Comment:
There's a fair bit of duplication between this procedure and the one above. Could be split out?

Reply:
Yes,  I'll cherry-pick the `_login_salt_api_sa` from 9d7e63d7d7f3e7601c0af02d2b1a61ae5ab73c82 (which I already customized in the 2.5 branch)

Reply:
Done (not a real cherry-pick, but backported the same logic/name)